group-actors/src/test/java/org/sunbird/actors/SearchGroupActorTest.java

```diff
@@ -0,0 +1,134 @@
+package org.sunbird.actors;
+
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+import akka.actor.ActorRef;
+import akka.actor.Props;
+import akka.testkit.javadsl.TestKit;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.sunbird.cassandra.CassandraOperation;
+import org.sunbird.cassandraimpl.CassandraOperationImpl;
+import org.sunbird.exception.BaseException;
+import org.sunbird.helper.ServiceFactory;
+import org.sunbird.message.Localizer;
+import org.sunbird.models.ActorOperations;
+import org.sunbird.request.Request;
+import org.sunbird.response.Response;
+import org.sunbird.util.JsonKey;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({Localizer.class, ServiceFactory.class})
+@PowerMockIgnore({"javax.management.*"})
+public class SearchGroupActorTest extends org.sunbird.actors.BaseActorTest {
+
+  private final Props props = Props.create(org.sunbird.actors.SearchGroupActor.class);
+  public static CassandraOperation cassandraOperation;
+
+  @Before
+  public void beforeEachTest() {
+    PowerMockito.mockStatic(Localizer.class);
+    when(Localizer.getInstance()).thenReturn(null);
+
+    PowerMockito.mockStatic(ServiceFactory.class);
+    cassandraOperation = mock(CassandraOperationImpl.class);
+    when(ServiceFactory.getInstance()).thenReturn(cassandraOperation);
+  }
+
+  @Test
+  public void testSearchGroup() {
```

Comment:
Am proposing to change these test names to make it meaningful. It is difficult to review.
SearchGroupActorTest
   - searchByIdReturnsXYZ
   - searchByInvalidIdReturnsClientError
   - searchWoUserTokenFails
   - searchWMuaTokenReturnsXYZ